### PR TITLE
clone non-primitives before returning result of .get()

### LINKED ISSUE #29: .get() returns a mutable internal object references. This is a bit confusing.
Here's what I'm talking about:

```javascript
var zc = require('zero-config');
var config = zc(process.cwd()); 

config.set('lol', { foo: { bar: 42 }});
console.log(config.get('lol'));
// {foo: { bar : 42 }}

config.get('lol').foo = 'mutated internally' // this is not good
console.log(config.get('lol'));
// { foo: 'mutated internally' }
```

I think that you should only be able to mutate the config via `.set()`. 
`.get()` should return copies/clones of objects, not internal references.

### PR DESCRIPTION
**Pull request by lxe**
Fixes #29

cc @Raynos

### CODE REVIEW

File: config-wrapper.js [hunk 1]
```diff
@@ -20,10 +20,20 @@ function ConfigWrapper(configObject, loose) {
...
-            return configObject;
+            return deepExtend({}, configObject);
         }
 
-        return getPath(configObject, keyPath);
+        // Clone non-primitives before returning
+        var val = getPath(configObject, keyPath);
+        if (typeof val === 'object') {
+            if (Array.isArray(val)) {
+                val = [].concat(val);
```
Comment (Raynos) [inline config-wrapper.js]: this means we dont deep clone arrays.

Comment (lxe) [inline config-wrapper.js]: You are correct. I'll just use deep-extend.

**Comment by anson627** [comment]
@lxe @Raynos this is probably what we want to do:
    function getKey(keyPath) {
        var clonedObject = deepExtend({}, configObject);
        if (!keyPath) {
            return clonedObject;
        }

        return getPath(clonedObject, keyPath);
    }

**Comment by Raynos** [comment]
@anson627 +1

**Pull request merged**